src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+
+const getPreparedPeople = (people: Person[]): Person[] => {
+  return people.map(person => {
+    return {
+      ...person,
+      mother: people.find(mother => mother.name === person.motherName),
+      father: people.find(father => father.name === person.fatherName),
+    };
+  });
+};
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isError, setIsError] = useState<boolean>(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setIsLoading(true);
+        const peopleFromServer = await getPeople();
+
+        setPeople(getPreparedPeople(peopleFromServer));
+      } catch (error) {
+        setIsError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
```

Comment:
I dont understand how to move this function fetchData outside the component PeoplePage.  Because setPeople, setIsLoading, setIsError working only inside of component. Because its useState. Can you help me with this point?